dbt_platform_helper/domain/maintenance_page.py

```diff
@@ -275,13 +278,35 @@ def __clean_up_maintenance_page_rules(
             listener_arn
         )
 
-        deletes = {}
+        # keep track of rules deleted
+        deletes = {"MaintenancePage": 0}
         for name in ["MaintenancePage", "AllowedIps", "BypassIpFilter", "AllowedSourceIps"]:
-            deleted = self.load_balancer.delete_listener_rule_by_tags(tag_descriptions, name)
-            deletes[name] = bool(deleted)
-            if fail_when_not_deleted and name == "MaintenancePage" and not deleted:
+            deleted_list = self.load_balancer.delete_listener_rule_by_tags(tag_descriptions, name)
+
+            # track the rules deleted grouped by service
+            for deleted_rule in deleted_list:
+                tags = {t["Key"]: t["Value"] for t in deleted_rule["Tags"]}
+                if tags.get("service"):
+                    if tags["service"] not in deletes:
+                        deletes[tags["service"]] = {
+                            "AllowedIps": 0,
+                            "BypassIpFilter": 0,
+                            "AllowedSourceIps": 0,
+                        }
+                    deletes[tags["service"]][name] += 1
+                elif tags.get("name") == "MaintenancePage":
+                    deletes["MaintenancePage"] += 1
+
+            if (
+                fail_when_not_deleted
+                and name == "MaintenancePage"
+                and deletes["MaintenancePage"] == 0
+            ):
                 raise ListenerRuleNotFoundException()
 
+        self.io.warn(
+            f"Rules deleted by type and grouped by service: {deletes}",
+        )
         return deletes
```

Comment:
Does this still need returning? Doesn't look like it's used any more

Reply:
No it doesn't, I can remove it for now as it isn't used.